react-carousel/src/components/Carousel.js

```diff
@@ -0,0 +1,309 @@
+        ? `${animationSpeed}ms, ${animationSpeed}ms`
+        : null,
+      transform: `translateX(${trackStyles.transform}px)`,
+      marginLeft: `${trackStyles.marginLeft}px`,
+      marginRight: `${trackStyles.marginRight}px`,
+    };
+
+    return (
+      <div className="BrainhubCarousel__trackContainer" ref={trackContainerRef}>
+        <ul
+          className={classnames('BrainhubCarousel__track', {
+            'BrainhubCarousel__track--transition': transitionEnabled,
+            'BrainhubCarousel__track--draggable': draggable,
+          })}
+          style={currentTrackStyles}
+          {...trackCustomProps}
+        >
+          {_compact(slides).map((carouselSlide, index) => (
+            <CarouselSlide
+              clickable
+              key={index}
+              currentSlideIndex={activeSlideIndex || props.value}
+              index={index}
+              width={props.itemWidth || itemWidth}
+              offset={index !== slides.length ? props.offset : 0}
+              onMouseDown={onMouseDown}
+              onTouchStart={onTouchStart}
+              isDragging={
+                Math.abs(slideMovement.dragOffset) > config.clickDragThreshold
+              }
+              itemClassNames={itemClassNames}
+              isDraggingEnabled={props.draggable}
+              {...slideCustomProps}
+            >
+              {carouselSlide}
+            </CarouselSlide>
+          ))}
+        </ul>
+      </div>
+    );
+  };
+
+  return (
+    <div className="BrainhubCarousel__container">
+      <div
+        className={classnames(
+          'BrainhubCarousel',
+          props.className,
+          ...(carouselClassNames || []),
+        )}
+        onMouseMove={onMouseMove}
+        ref={carouselRef}
+        {...carouselCustomProps}
+      >
+        {React.createElement(React.Fragment, null, beforeCarouselItems)}
+        {renderCarouselItems()}
+        {React.createElement(React.Fragment, null, afterCarouselItems)}
+      </div>
+    </div>
+  );
+};
+
+Carousel.propTypes = {
+  itemWidth: PropTypes.number,
+  width: PropTypes.number,
+  value: PropTypes.number,
+  onChange: PropTypes.func,
+  children: PropTypes.node,
+  slides: PropTypes.arrayOf(PropTypes.node),
+  offset: PropTypes.number,
+  draggable: PropTypes.bool,
+  animationSpeed: PropTypes.number,
+  className: PropTypes.string,
+  transformOffset: PropTypes.number,
+  nearestSlideIndex: PropTypes.number,
+  plugins: PropTypes.arrayOf(
+    PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.func,
+      PropTypes.shape({
+        resolve: PropTypes.func,
+        options: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+      }),
+    ]),
+  ),
+  breakpoints: PropTypes.objectOf(
+    PropTypes.shape({
+      slidesPerPage: PropTypes.number,
+      draggable: PropTypes.bool,
+      animationSpeed: PropTypes.number,
+      dots: PropTypes.bool,
+      className: PropTypes.string,
+      transformOffset: PropTypes.string,
```

Comment:
shouldn't it be a number?

Reply:
all props for breakpoints were outdated. changed